hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,89 @@
+
+## Features
+- Provides support for both deterministic (DRNG) and true random number generation (TRNG), when combined with a secure entropy source (i.e. one constructed and implemented in compliance with SP 800-90A,B,C and AIS31).
+- Compliant with NIST and BSI recommendations for random number generation
+- In deterministic mode, meets the requirements given in AIS31 for a DRG.3 class deterministic random number generator (DRNG) meaning it provides Forward Secrecy and Enhanced Backward Secrecy.
+- Utilizes the CTR_DRBG construction specified in NIST SP 800-90A, qualifying it as an NIST-approved DRBG (deterministic random bit generator).
+    - Operates at 256bit security strength.
+- Support for multiple separate CSRNG instances per IP block.
+    - Each instance has its own internal state, control, reseed counters and IO pins.
+    - The number of RNG instances is set via a module parameter
+- Software access to a dedicated RNG instance.
+    - One instance, Instance 0, is always accessible through TLUL registers,
+    - All other instances route to other hardware peripherals (e.g. the key manager, obfuscation engines etc.) and in normal operation these other instances are inaccessible from software.
+    - The IP may be configured to support “debug mode” wherein all instances can be accessed by software.
+      For security reasons this mode may be permanently disabled using fuses.
+- The IP interfaces with external entropy sources to obtain any required non-determinstic seed material (entropy) and nonces.
+    - Requires an external entropy source which is compliant with NIST SP 800-90B, and which also satisfies the requirements for a PTG.2 class physical non-deterministic random number generator as defined in AIS31.
+    - Dedicated hardware interface with external entropy source satisfies requirements for get_entropy_input() interface as defined in SP 800-90A.
+    - When needed, utilizes the Block_Cipher_df derivation function (as defined in SP 800-90A) for assembling seed material.
+      This allows the use of entropy sources which are not full-entropy (less than one bit of entropy per bit generated).
+- Also supports the optional use of personalization strings or other application inputs (e.g. OTP memory values) during instantiation.
+- Assuming a continuously-live entropy source, each instance can also optionally be used as a non-determinstic TRNG (true random number generator--also called a  non-deterministic random bit generator or NRBG in SP 800-90C).
+    - In this mode, an instance also meets the requirements laid out for a PTG.3 class RNG, the strongest class laid out in AIS31.
+    - Implementation follows the NRBG “Oversampling Construction” approved by SP 800-90C, to meet both CC and FIPS TRNG constructions.
+- In addition to the approved DRNG node, any instance can also operate in “Fully Deterministic mode”, meaning the seed depends entirely on application inputs or personalization strings.
```

Comment:
node -> mode? (Not sure, could be a special term?)

Reply:
Changed. Thanks for spotting that.